Traceback (most recent call last):
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Users\denis.osses\Anaconda3\lib\asyncio\base_events.py", line 647, in run_until_complete
    return future.result()
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\denis.osses\Anaconda3\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import solve_ivp

# Par√°metros de la ley de enfriamiento de Newton
k = -0.1  # Constante de enfriamiento

# Funci√≥n que define la temperatura del medio variable en el tiempo
def T_m(t):
    return 20 + 10 * np.sin(0.1 * t)

# Definici√≥n de la EDO para la ley de enfriamiento de Newton
def Newton(t, T):
    return k * (T - T_m(t))

# Condici√≥n inicial
T0 = 100  # Temperatura inicial del objeto

# Intervalo de tiempo para la simulaci√≥n
t_span = (0, 100)
t_eval = np.linspace(t_span[0], t_span[1], 1000)

# Resolver la EDO
sol = solve_ivp(Newton, t_span, [T0], t_eval=t_eval)

# Graficar los resultados
plt.plot(sol.t, sol.y[0], label='Temperatura del objeto')
plt.plot(sol.t, T_env(sol.t), label='Temperatura del medio', linestyle='dashed')
plt.xlabel('Tiempo')
plt.ylabel('Temperatura')
plt.legend()
plt.title('Ley de Enfriamiento de Newton con Temperatura del Medio Variable')
plt.show()
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mNameError[0m                                 Traceback (most recent call last)
[1;32m~\AppData\Local\Temp\ipykernel_45744\721938758.py[0m in [0;36m<module>[1;34m[0m
[0;32m     26[0m [1;31m# Graficar los resultados[0m[1;33m[0m[1;33m[0m[0m
[0;32m     27[0m [0mplt[0m[1;33m.[0m[0mplot[0m[1;33m([0m[0msol[0m[1;33m.[0m[0mt[0m[1;33m,[0m [0msol[0m[1;33m.[0m[0my[0m[1;33m[[0m[1;36m0[0m[1;33m][0m[1;33m,[0m [0mlabel[0m[1;33m=[0m[1;34m'Temperatura del objeto'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[1;32m---> 28[1;33m [0mplt[0m[1;33m.[0m[0mplot[0m[1;33m([0m[0msol[0m[1;33m.[0m[0mt[0m[1;33m,[0m [0mT_env[0m[1;33m([0m[0msol[0m[1;33m.[0m[0mt[0m[1;33m)[0m[1;33m,[0m [0mlabel[0m[1;33m=[0m[1;34m'Temperatura del medio'[0m[1;33m,[0m [0mlinestyle[0m[1;33m=[0m[1;34m'dashed'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0m[0;32m     29[0m [0mplt[0m[1;33m.[0m[0mxlabel[0m[1;33m([0m[1;34m'Tiempo'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0;32m     30[0m [0mplt[0m[1;33m.[0m[0mylabel[0m[1;33m([0m[1;34m'Temperatura'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m

[1;31mNameError[0m: name 'T_env' is not defined
NameError: name 'T_env' is not defined

